File: app/src/androidTest/java/info/sanaebadi/tddandroidunittest/NoteDatabaseTest.java
package info.sanaebadi.tddandroidunittest;

import androidx.room.Room;
import androidx.test.core.app.ApplicationProvider;


import org.junit.After;
import org.junit.Before;

import info.sanaebadi.tddandroidunittest.persistence.NoteDao;
import info.sanaebadi.tddandroidunittest.persistence.NoteDatabase;

public abstract class NoteDatabaseTest {

    // system under test
    private NoteDatabase noteDatabase;


    public NoteDao getNoteDao(){
        return noteDatabase.getNoteDao();
    }

    @Before
    public void init(){
        noteDatabase = Room.inMemoryDatabaseBuilder(
                ApplicationProvider.getApplicationContext(),
                NoteDatabase.class
        ).build();
    }

    @After
    public void finish(){
        noteDatabase.close();
    }
}

File: app/src/androidTest/java/info/sanaebadi/tddandroidunittest/NoteDatabaseTest.kt
package info.sanaebadi.tddandroidunittest

import androidx.room.Room
import androidx.test.core.app.ApplicationProvider
import info.sanaebadi.tddandroidunittest.persistence.NoteDao
import info.sanaebadi.tddandroidunittest.persistence.NoteDatabase
import org.junit.After
import org.junit.Before

abstract class NoteDatabaseTest {

    var noteDatabase: NoteDatabase? = null

    fun getNoteDao(): NoteDao {
        return noteDatabase!!.gtNoteDao()
    }


    @Before
    fun init() {
        noteDatabase = Room.inMemoryDatabaseBuilder(
            ApplicationProvider.getApplicationContext(),
            NoteDatabase::class.java
        ).build()
    }

    @After
    fun finish(){
        noteDatabase!!.close()
    }
}